# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/ServiceWorkerRegistration.webidl
# Commit: afae3c50f7ac
# Full Hash: afae3c50f7acff60d9400adbdc3a8aa2f9d7bdfe
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-06-10 15:47:45
# Regressor Bug: 1564235
# File Overlap Count: 3
# Description:
#   Bug 1564235 - P2 Create NavigationPreloadManager.webidl and implement webidl binding. r=dom-worker-reviewers,webidl,asuth
#   
#   This patch creates NavigationPreloadManager.webidl and implements the webidl binding.
#   
#   Depends on D116913
# ==============================================================================

diff -r 2148ef30186a -r afae3c50f7ac dom/webidl/ServiceWorkerRegistration.webidl
--- a/dom/webidl/ServiceWorkerRegistration.webidl	Thu Jun 10 06:01:26 2021 +0000
+++ b/dom/webidl/ServiceWorkerRegistration.webidl	Thu Jun 10 06:01:26 2021 +0000
@@ -16,6 +16,9 @@
   readonly attribute ServiceWorker? waiting;
   readonly attribute ServiceWorker? active;
 
+  [Pref="dom.serviceWorkers.navigationPreload.enabled", SameObject]
+  readonly attribute NavigationPreloadManager navigationPreload;
+
   readonly attribute USVString scope;
   [Throws]
   readonly attribute ServiceWorkerUpdateViaCache updateViaCache;